p2p/service.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABC, abstractmethod
+import asyncio
+import logging  # noqa: F401
+from typing import Callable, Optional
+
+from p2p.cancel_token import CancelToken
+from p2p.exceptions import OperationCancelled
+
+
+class BaseService(ABC):
+    # Subclasses must define their own logger.
+    logger = None  # type: logging.Logger
```

Comment:
We can use python 3.6 syntax for these types.

```python
logger: logging.Logger = None
```

Reply:
I'm with you but I don't think it's fair to do it anytime in the near term future.  We *just* forced everyone to upgrade to 3.x and while `3.5 -> 3.6` is not a significant as `2.x -> 3.x` it's still not something we can just throw at the community imho.  I'm sure that there are people for whom the upgrade would require some non-trivial amount of infrastructure upgrades to get all of their ops infrastructure in a 3.6 environment.

Reply:
That would also allow ethpm to start with 3.6. But I can't say much about the community backing of such a move.